### ISSUE TITLE
[LINT_BUG]: Function argument confused with a box module

### ISSUE THREAD

**Issue by Gotfrid** [Appsilon/box.linters#131]
### box.linters version

```
0.9.1
```

### Sample source code to lint

```R
box::use(
  app/logic/table,
)

#' @param data list(summary = reactive(data.frame()))
#' @export
do_something <- function(data) {
  table$summary(data$summary())
}
```

### Lint command used

```
rhino::lint_r('app.R')
```

### Lint result

```
app.R:8:17: warning: [box_usage_linter] <package/module>$function does not exist.
  table$summary(data$summary())
                ^~~~~~~~~~~~
```

### Expected result

No errors expected. Note that this behaviour happens specifically when there is a box::use statement that imports a local module.

**Comment by radbasa** [comment]
This is a good catch!

We have some logic to handle the `object$function()` pattern. We identify if the pair exists as exported by a `box::use` call, or it is a special use case as in the example code.

However, it now sees `data$summary()` and trips over `summary()` that exists as part of `table$summary()`. Will need to rethink the logic for this.
